hypothesis-python/src/hypothesis/extra/redis.py

```diff
@@ -0,0 +1,75 @@
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis/
+#
+# Most of this work is copyright (C) 2013-2020 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from contextlib import contextmanager
+from datetime import timedelta
+from typing import Iterable
+
+from redis import Redis
+
+from hypothesis.database import ExampleDatabase
+from hypothesis.internal.validation import check_type
+
+
+class RedisExampleDatabase(ExampleDatabase):
+    """Store Hypothesis examples as sets in the given :class:`redis.Redis` datastore.
+
+    This is particularly useful for shared databases, as per the recipe
+    for a :class:`~hypothesis.database.MultiplexedDatabase`.
+    If a test has not been run for ``expire_after``, those examples will be allowed
+    to expire.  The default time-to-live persists examples between weekly runs.
+    """
+
+    _prefix: bytes = b"hypothesis-example:"
+
+    def __init__(self, redis: Redis, *, expire_after: timedelta = timedelta(days=8)):
```

Comment:
Why 8? Just curious...

Reply:
Also, why expiration logic is backend specific? Shouldn't it be more general and backend would just implement required routines to maintain it?

Reply:
Eight because I was thinking about weekly cron jobs - and you don't want expire keys a few seconds before they're accessed.

Expiration logic is Redis-specific because it's more difficult to clear out Redis keys by age than delete local directories, and because unlike the others there's a non-Hypothesis server responsible for expiration.

Reply:
After some thoughts, I agree with that.  How about to make this behaviour more screaming? Like, with `..note` / `..warning` sphinx things? I was trying to write about "let's document this" at first place, but after third iteration of reading code I noticed "weekly" word - suddenly, for nonnative speaker this may not be a stop word to realize some important behaviour. Personal experience, but else I don't have. 

In the end, this behavior makes this backend different from others by ability to remove old data. This may be not always good. Imagine you went to Hawaii on vacation for two weeks and after return all your examples were vanished while no other developer ran any test (holidays!).

Pardon me, may be I just too strict about the thing that would delete some data by default without explicit asking about. May be another way would be to not expire anything at all, but let users to decide whenever they want do this and how? Much like with the other backends.

Reply:
Note that it's not *old* data, it's *unaccessed* data - and cross-platform `atime` is a mess.

I'll make the expiry note much more obvious though.

Reply:
That would be a good clarification and it removes all my doubts. Thanks!